types/mui-datatables/index.d.ts

```diff
@@ -138,7 +139,13 @@ export interface MUIDataTableFilterOptions {
      * `filterList` must be of the same type in the main column options, that is an array of arrays, where each array corresponds to the filter list for a given column.
      * {@link https://github.com/gregnb/mui-datatables/blob/master/examples/customize-filter/index.js Example}
      */
-    display?: (filterList: string[], onChange: any, index: number, column: any) => void;
+    display?: (
+        filterList: MUIDataTableState['filterList'],
+        onChange: (val: string | string[], index: number, column: MUIDataTableColumn) => void,
```

Comment:
`onChange` is passed [`handleCustomChange()`](https://github.com/gregnb/mui-datatables/blob/master/src/components/TableFilter.js#L151), which can receive either a string or array of strings